akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -499,12 +495,17 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
       byteBuffer.putShort(length.toShort)
       ensureLiteralCharsLength(length)
       val bytes = literalBytes
-      val chars = Unsafe.instance.getObject(literal, StringValueFieldOffset).asInstanceOf[Array[Char]]
-      var i = 0
-      while (i < length) {
-        // UsAscii
-        bytes(i) = chars(i).asInstanceOf[Byte]
-        i += 1
+      if (Unsafe.javaVersion > 8) {
```

Comment:
@ktoso @patriknw a hack to compensate another hack, not pretty but should get the job done without losing performance, in fact; in Java 9 because of `System.arraycopy` and no casting back to `char` it should be slightly faster.